ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/commands/CommandCompletionServiceIT.scala

```diff
@@ -154,6 +162,64 @@ class CommandCompletionServiceIT
             commandIds3 should not contain (commandIds1(1))
           }
       }
+
+      "implicitly tail the stream if no offset is passed" in allFixtures { ctx =>
+        // A command to create a Test.Dummy contract (see //ledger/sandbox/src/main/resources/damls/Test.daml)
+        val createDummyCommand = ProtoCommand(
+          ProtoCommand.Command.Create(
+            ProtoCreateCommand(
+              templateId = Some(templateIds.dummy),
+              createArguments = Some(
+                Record(
+                  fields = Seq(
+                    RecordField("operator", Some(Value(Value.Sum.Party(party))))
+                  )
+                ))
+            )))
+
+        // Issue the `createDummyCommand` to the ledger using the managed command client
+        def createDummy(id: String) =
+          for {
+            commandClient <- ctx.commandClient(ctx.ledgerId)
+            request = ctx.command(id, Seq(createDummyCommand))
+            _ <- commandClient.submitSingleCommand(request)
+          } yield ()
+
+        // Issues several `createDummyCommand`s
+        def createDummies(ids: List[String]) =
+          Future.sequence(ids.map(createDummy)).map(_ => ())
+
+        // Starts "tailing" the completions service and then issues one more command
+        def tailCompletionsAndConcurrentlyCreateDummyWithCommandId(id: String) = {
+          val completions = completionSource(
+            ctx.commandCompletionService,
+            ctx.ledgerId,
+            applicationId,
+            Seq(party),
+            offset = None).runWith(Sink.queue())
+          createDummy(id)
+          completions
+        }
+
+        // Issue 10 create commands, then listen from the ledger end (implicitly by passing no offset) and
+        // verify that the first received completion is for the last created contract
+
+        val arbitraryCommandIds = List.tabulate(10)(_.toString)
+        val targetCommandId = "the-one"
+
+        for {
+          () <- createDummies(arbitraryCommandIds)
```

Comment:
I mostly use `_` in such cases, didn't know that `()` also works. What's the difference?

Reply:
Using `()` means that you're actually pattern matching on the returned `Unit` value.
@S11001001: Will this be desugared to `withFilter` due to the pattern matching?

Reply:
It's mostly to make sure there is a compilation error is some type change. I'm not sure it's worth spending time caring about this detail, I believe the impact of this in the integration tests running time would barely be perceivable.

Reply:
I slightly changed the helper functions return types and changed this for convenience.

Reply:
@gerolf-da Yes.

@stefanobaghino-da That means you don't quite get the type-safety you expect:

```scala
scala> for (() <- List(1:Any, 2)) yield 42
res1: List[Int] = List()
```

Instead:

```suggestion
          _ <- createDummies(arbitraryCommandIds): Future[Unit]
```

Reply:
Addressed by https://github.com/digital-asset/daml/pull/1961/commits/1f8fae9ab62c717e1424427319070ee4e186564f